diff -rupN freetype-2.5.3/builds/unix/config.sub freetype-2.5.3-ghost/builds/unix/config.sub
--- freetype-2.5.3/builds/unix/config.sub	2014-03-06 23:13:51.060164300 +0100
+++ freetype-2.5.3-ghost/builds/unix/config.sub	2014-09-03 15:02:31.014256000 +0200
@@ -1356,7 +1356,7 @@ case $os in
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
-	      | -aos* | -aros* \
+	      | -aos* | -aros* | -ghost* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
