diff -Nur cairo-1.12.18/build/config.sub cairo-1.12.18-ghost/build/config.sub
--- cairo-1.12.18/build/config.sub	2020-02-18 14:43:52.844132000 +0100
+++ cairo-1.12.18-ghost/build/config.sub	2020-02-18 14:44:10.038776400 +0100
@@ -1348,7 +1348,7 @@
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
-	      | -aos* | -aros* \
+	      | -aos* | -aros* | -ghost* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
