diff -Nur pixman-0.38.0/config.sub pixman-0.38.0-ghost/config.sub
--- pixman-0.38.0/config.sub	2020-02-18 11:19:59.751974600 +0100
+++ pixman-0.38.0-ghost/config.sub	2020-02-18 11:20:13.965683000 +0100
@@ -1368,7 +1368,7 @@
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
-	      | -aos* | -aros* | -cloudabi* | -sortix* \
+	      | -aos* | -aros* | -cloudabi* | -sortix* | -ghost* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -knetbsd* | -mirbsd* | -netbsd* \
diff -Nur pixman-0.38.0/configure pixman-0.38.0-ghost/configure
--- pixman-0.38.0/configure	2020-02-18 11:20:00.220557600 +0100
+++ pixman-0.38.0-ghost/configure	2020-02-18 11:23:26.751692900 +0100
@@ -5660,6 +5660,10 @@
   fi
   ;;
 
+ghost*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 haiku*)
   lt_cv_deplibs_check_method=pass_all
   ;;
@@ -11246,6 +11250,19 @@
   esac
   ;;
 
+ghost*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  dynamic_linker="ghostld"
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  shlibpath_var=LIBRARY_PATH
+  shlibpath_overrides_runpath=no
+  sys_lib_dlsearch_path_spec='/system/lib'
+  hardcode_into_libs=yes
+  ;;
+
 haiku*)
   version_type=linux # correct to gnu/linux during the next big refactor
   need_lib_prefix=no
